src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
                 '/home/{}/.ssh/authorized_keys'.format(namespace.admin_username)
 
 
+def _validate_admin_username(username, is_linux):
+    if not username:
+        return "admin user name can not be empty"
+    # pylint: disable=line-too-long
+    pattern = (r'[\\\/"\[\]:|<>+=;,?*@#()!A-Z]+' if is_linux else r'[\\\/"\[\]:|<>+=;,?*@]+')
+    linux_err = r'admin user name cannot contain upper case character A-Z, special characters \/"[]:|<>+=;,?*@#()! or start with $ or -'
+    win_err = r'admin user name cannot contain special characters \/"[]:|<>+=;,?*@# or ends with .'
+    if re.findall(pattern, username):
+        raise CLIError(linux_err if is_linux else win_err)
+    if is_linux and re.findall(r'^[$-]+', username):
+        raise CLIError(linux_err)
+    if not is_linux and username.endswith('.'):
+        raise CLIError(win_err)
+    disallowed_user_names = [
+        "administrator", "admin", "user", "user1", "test", "user2",
+        "test1", "user3", "admin1", "1", "123", "a", "actuser", "adm",
+        "admin2", "aspnet", "backup", "console", "david", "guest", "john",
+        "owner", "root", "server", "sql", "support", "support_388945a0",
+        "sys", "test2", "test3", "user4", "user5"]
+    if username.lower() in disallowed_user_names:
+        raise CLIError('The specified admin user name is not allowed, as it uses reserved words. Try again with a different value')
```

Comment:
Reserved words sounds like programming keywords like int, char, etc. Plus it's hard to say that "support_388945a0" is reserved. 😁  I would recommend something like "This user name "foo" meets the general requirements, but is specifically disallowed for this image. Please try a different value."